### PR TITLE
Complete Space Travelers' Hub website

### PR DESCRIPTION
**Pull request by Rudy-Menekam**
1. It consists of Rockets, Missions, and the My Profile section.
2. Fetch data and update the Redux store
3. Use useSelector() Redux Hook to select the state slices and render lists of rockets and missions in corresponding routes
4. Write actions and reducers for booking rockets and joining missions
5. Write actions and reducers for canceling rockets and leaving missions
6. Render UI: conditional components rendering
7. Render UI: My Profile section

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
+import Rockets from './components/rockets';
+import MyProfile from './components/my_profile';
+import Header from './components/header';
+import { getMission } from './reduxState/missionSlice';
 
 function App() {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(getMission());
+  }, [dispatch]);
```
Comment (julie-ify) [inline src/App.js:15]: - [ ] Right now, when the app opens the missions are fetched. The missions should not be fetched until we visit the mission component. Kindly fix that.

File: src/reduxState/missionSlice.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      const missions = state.missions.filter(
+        (mission) => mission.reserved === true,
+      );
+      return { ...state, reserved: missions };
+    },
+  },
+  extraReducers: (builder) => {
+    builder
+      .addCase(getMission.fulfilled, (state, action) => {
+        state.missions = action.payload;
```
Comment (julie-ify) [inline src/reduxState/missionSlice.js:45]: - [ ] Let's filter the mission attributes based on the listed values in the project requirement. You are saving all the values from the store but that is not correct. Only the listed values from the requirement should be stored in the store. Kindly fix. You can take a clue from the rocket's state to filter out unnecessary values.

  iorro

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Rudy-Menekam and @chepkok3,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Filtered values for the rockets ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @Rudy-Menekam and @chepkok3,

## APPROVED 🟢

## Highlights
✔️ Filter of API data

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**